src/prioritized_pool/tests.rs

```diff
@@ -0,0 +1,281 @@
+use {
+	super::*,
+	crate::alloy::primitives::Address,
+	std::{collections::HashMap, convert::Infallible},
+};
+
+struct PrioritizedOrderpoolHashMapNonces(HashMap<Address, u64>);
+
+impl From<HashMap<Address, u64>> for PrioritizedOrderpoolHashMapNonces {
+	fn from(map: HashMap<Address, u64>) -> Self {
+		Self(map)
+	}
+}
+
+impl PrioritizedOrderpoolNonceSource for PrioritizedOrderpoolHashMapNonces {
+	type NonceError = Infallible;
+
+	fn nonce(&self, address: &Address) -> Result<u64, Self::NonceError> {
+		Ok(self.0.get(address).copied().unwrap_or_default())
+	}
+}
+
+#[derive(Clone)]
+pub struct PrioritizedOrderpoolTestBundle {
+	id: u64,
+	nonces: Vec<BundleNonce>,
+	profit: u64,
+}
+
+impl PrioritizedOrderpoolOrder for PrioritizedOrderpoolTestBundle {
+	type ID = u64;
+
+	fn id(&self) -> Self::ID {
+		self.id
+	}
+
+	fn nonces(&self) -> Vec<BundleNonce> {
+		self.nonces.clone()
+	}
+}
+
+type PrioritizedOrderpoolTestPriority = u64;
+
+impl PrioritizedOrderpoolPriority for PrioritizedOrderpoolTestPriority {
+	type Order = PrioritizedOrderpoolTestBundle;
+
+	fn new(order: &Self::Order) -> Self {
+		order.profit
+	}
+}
+
+/// Helper struct for common `PrioritizedOrderStore` test operations
+/// Works hardcoded on max profit ordering since it changes nothing on internal
+/// logic
+struct TestContext {
+	generated_id_count: u64,
+	nonces: PrioritizedOrderpoolHashMapNonces,
+	order_pool: PrioritizedOrderpool<
+		PrioritizedOrderpoolTestPriority,
+		PrioritizedOrderpoolTestBundle,
+	>,
+}
```

Comment:
There is also a test utility called `OneStep` that you can use to unit test a single step in more realistic setting as if it was running in a pipeline and responding to a payload job. Here's an example of how it is used:

https://github.com/flashbots/rblib/blob/f4abbf12883e4486d6a3898d8b00613dec444618/src/steps/revert.rs#L300-L313